contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+     */
+    function setMinMaxNotional(
+        address _underlying,
+        uint256 _min,
+        uint256 _max
+    ) external onlyOwner {
+        require(_underlying != address(0), "OTCWrapper: asset address cannot be 0");
+        require(_min > 0, "OTCWrapper: minimum notional cannot be 0");
+        require(_max > 0, "OTCWrapper: maximum notional cannot be 0");
+
+        minMaxNotional[_underlying].min = _min;
+        minMaxNotional[_underlying].max = _max;
+    }
+
+    /**
+     * @notice sets the whitelist status of market maker addresses
+     * @dev can only be called by owner
+     * @param _marketMakerAddress address of the market maker
+     * @param _isWhitelisted bool with whitelist status
+     */
+    function setWhitelistMarketMaker(address _marketMakerAddress, bool _isWhitelisted) external onlyOwner {
+        require(_marketMakerAddress != address(0), "OTCWrapper: market maker address cannot be 0");
+        require(
+            isWhitelistedMarketMaker[_marketMakerAddress] != _isWhitelisted,
+            "OTCWrapper: whitelist status already in place"
+        );
+
+        isWhitelistedMarketMaker[_marketMakerAddress] = _isWhitelisted;
+    }
+
+    /**
+     * @notice sets the fee for a given the underlying asset
+     * @dev can only be called by owner
+     * @param _underlying underlying asset address
+     * @param _fee fee amount in bps (100 = 1%)
+     */
+    function setFee(address _underlying, uint256 _fee) external onlyOwner {
+        require(_underlying != address(0), "OTCWrapper: asset address cannot be 0");
+        require(_fee > 0, "OTCWrapper: fee cannot be 0");
+
+        fee[_underlying] = _fee;
+    }
+
+    /**
+     * @notice sets the beneficiary address
+     * @dev can only be called by owner
+     * @param _beneficiary beneficiary address
+     */
+    function setBeneficiary(address _beneficiary) external onlyOwner {
+        require(_beneficiary != address(0), "OTCWrapper: beneficiary address cannot be 0");
+
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice sets the fill deadline duration
+     * @dev can only be called by owner
+     * @param _fillDeadline fill deadline duration
+     */
+    function setFillDeadline(uint256 _fillDeadline) external onlyOwner {
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+
+        fillDeadline = _fillDeadline;
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice allows market maker to deposit collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to deposit
+     */
+    function depositCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
```

Comment:
we should also check whether order succeeded. we can kill two birds w one stone with

`require(orderStatus[_orderID] = succeeded, "OTCWrapper: inexistent or unsuccessful order");`

Reply:
true. could have done either way